File: server/02_BOL/OrderModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace _02_BOL
{
    public class OrderModel
    {
        public int OrderID { get; set; }

        [Required, DisplayFormat(DataFormatString = "{0:MMM dd, yyyy}")]
        public DateTime RentalStart { get; set; }

        [Required, DisplayFormat(DataFormatString = "{0:MMM dd, yyyy}")]
        public DateTime RentalEnd { get; set; }

        [DisplayFormat(DataFormatString = "{0:MMM dd, yyyy}")]
        public Nullable<DateTime> ActualRentalEnd { get; set; }

        [Required]
        public int UserID { get; set; }

        [Required]
        public int CarID { get; set; }
    }
}

File: client/src/app/shared/models/order.model.ts
export class Order {

    OrderID?: number;

    RentalStart: any; // the reason it is 'any' and not 'Date' is for preview it as default value for 'input type date'
                     // which accept only 'yyyy-MM-dd' format that can be only in type string - (in 'order-view-manager') 
    RentalEnd: any;

    ActualRentalEnd: any;

    UserID: number;

    CarID: number;
}